cpp/src/parquet/encryption/crypto_factory.h

```diff
@@ -148,7 +146,7 @@ class PARQUET_EXPORT CryptoFactory {
       int dek_length, const std::string& column_keys, FileKeyWrapper* key_wrapper);
 
   /// Key utilities object for kms client initialization and cache control
-  KeyToolkit key_toolkit_;
+  std::shared_ptr<KeyToolkit> key_toolkit_ = std::make_shared<KeyToolkit>();
```

Comment:
Is it desirable for this to be default-constructed? I suppose concrete use requires a particular key toolkit, not a default-constructed one?

Reply:
I think this is fine. The `key_toolkit_` is always left as the default constructed value, and it is initialized when `CryptoFactory::RegisterKmsClientFactory` is called, which in turn calls `key_toolkit_->RegisterKmsClientFactory`.